modules/reference/pages/tunable-properties.adoc

```diff
@@ -1372,6 +1372,36 @@ Maximum size of the user-space receive buffer. If `null`, this limit is not appl
 
 ---
 
+[[kafka_throughput_throttling_v2]]
+=== kafka_throughput_throttling_v2
+
+Enables an updated throttling algorithm based on a shared token bucket, introduced with Redpanda v23.3.8. Set this property to false if you need to use the quota balancing algorithm from Redpanda v23.3.7 and older.  This property defaults to true for all new or upgraded Redpanda clusters.
+
+*Type*: boolean
+
+*Default*: true
+
+*Restart required*: true
+
+WARNING: Disabling this property causes your Redpanda cluster to use an outdated throughput throttling mechanism. Only set this to `false` when advised to do so by Redpanda support.
+
+---
+
+[[kafka_throughput_replenish_threshold]]
+=== kafka_throughput_replenish_threshold
+
+Threshold for refilling the token bucket. The range for this threshold is automatically clamped to the corresponding throughput limit for ingress and egress.
```

Comment:
@BenPope  does this mean: 'increments of throughput in which throughput throttling is re-evaluated' trying to translate to user-speak here.

Reply:
> @BenPope does this mean: 'increments of throughput in which throughput throttling is re-evaluated' trying to translate to user-speak here.

It's reevaluated every request, but the number of tokens wouldn't be updated if the amount to replenish is less than the threshold. This allows avoiding the atomic write of the token count on every request.